src/org/zaproxy/zap/extension/spider/SpiderThread.java

```diff
@@ -589,4 +630,8 @@ public void setCustomParseFilters(List<ParseFilter> customParseFilters) {
 		this.customParseFilters  = customParseFilters;
 	}
 
+    public int getNumberOfNodesAdded() {
+        return getAddedNodesTableModel().getRowCount();
```

Comment:
Maybe there's nothing wrong with this but it just seems strange to me. Shouldn't the spider actually keep track of this vs the table model row count?

Since this is done in a thread class can we end up with a sync or time of check vs time of use issue?